### PR TITLE
feat: remove cache file after deploy & fix oom in zip progress

### PR DESCRIPTION
**Pull request by Siglud**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_sprints/taskboard/Compute%20Team/Microsoft%20Teams%20Extensibility/Germanium/CY23Q4/2Wk/2Wk03%20(Oct%2022%20-%20Nov%2004)?workitem=25518805

### CODE REVIEW

File: packages/fx-core/src/component/driver/deploy/azure/impl/azureDeployImpl.ts [hunk 1]
```diff
@@ -89,6 +91,27 @@ export abstract class AzureDeployImpl extends BaseDeployImpl {
...
+   * cleanup function after deployment is finished
+   * @protected
+   */
+  protected async cleanup(): Promise<void> {
+    if (this.zipFilePath && !this.dryRun && fs.existsSync(this.zipFilePath)) {
+      try {
+        await fs.remove(this.zipFilePath);
+        // if upper folder is empty, remove it
+        const parentFolder = path.dirname(this.zipFilePath);
+        if ((await fs.readdir(parentFolder)).length === 0) {
```
Comment (hund030) [inline packages/fx-core/src/component/driver/deploy/azure/impl/azureDeployImpl.ts:104]: It seems that the `zipFilePath` can be entered by the user. Removing the parent folder may also delete unexpected files. 
For instance, if the user specifies the zip file path as `C:/deployment.zip`, will this code remove all files from disk C?
https://github.com/OfficeDev/TeamsFx/wiki/Available-actions-in-Teams-Toolkit#azurefunctionszipdeploy

**Review by hund030 — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -661,6 +661,8 @@
   "error.arm.ConvertArmOutputError": "Unable to convert ARM deployment result to action output, there is a duplicated key '%s' in ARM deployment result.",
   "error.deploy.DeployEmptyFolderError": "Unable to locate any files in the distribution folder: '%s'. Please ensure that the folder is not empty and that all necessary files have been included.",
   "error.deploy.CheckDeploymentStatusTimeoutError": "Unable to check deployment status because the process timed out. Check your internet connection and try again. If the issue persists, please review the deployment logs (Deployment -> Deployment center -> Logs) in Azure portal to identify any issues that may have occurred.",
+  "error.deploy.ZipFileError": "Unable to zip the distribution folder. Please ensure that the folder is not larger then 2GB.",
```
Comment (yukun-dong) [inline packages/fx-core/resource/package.nls.json:664]: typo: "than". Also curious about the meaning of "distribution folder".

**Review by huimiu — APPROVED** [review]

File: packages/fx-core/resource/package.nls.json (hunk 2 above)
Comment (Siglud) [reply packages/fx-core/resource/package.nls.json:664]: nice catch, changed to artifact folder to fit with the YAML file.

**Review by yukun-dong — DISMISSED** [review]

**Review by LongOddCode — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -661,6 +661,8 @@
   "error.arm.ConvertArmOutputError": "Unable to convert ARM deployment result to action output, there is a duplicated key '%s' in ARM deployment result.",
   "error.deploy.DeployEmptyFolderError": "Unable to locate any files in the distribution folder: '%s'. Please ensure that the folder is not empty and that all necessary files have been included.",
   "error.deploy.CheckDeploymentStatusTimeoutError": "Unable to check deployment status because the process timed out. Check your internet connection and try again. If the issue persists, please review the deployment logs (Deployment -> Deployment center -> Logs) in Azure portal to identify any issues that may have occurred.",
+  "error.deploy.ZipFileError": "Unable to zip the artifact folder. Please ensure that the folder is not larger than 2GB.",
+  "error.deploy.ZipFileTargetInUse": "Unable to clear the distribution zip file. Please ensure that the file is not in use.",
```
[the remark this answers is not in the document]
Comment (Siglud) [reply packages/fx-core/resource/package.nls.json]: Sure! Added zip path to the error message.

**Review by therealjohn — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by yukun-dong — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Pull request merged**